src/lib.rs

```diff
@@ -98,104 +85,96 @@ pub enum Base {
 
 impl Base {
     /// Get the base code.
-    pub fn code(&self) -> &str {
+    pub fn code(&self) -> u8 {
         use Base::*;
 
         match *self {
-            Base1 => "1",
-            Base2 => "0",
-            Base8 => "7",
-            Base10 => "9",
-            Base16 => "f",
-            Base16Upper => "F",
-            Base32hex => "v",
-            Base32hexUpper => "V",
-            Base32hexpad => "t",
-            Base32hexpadUpper => "T",
-            Base32 => "b",
-            Base32Upper => "B",
-            Base32pad => "c",
-            Base32padUpper => "C",
-            Base32z => "h",
-            Base58flickr => "Z",
-            Base58btc => "z",
-            Base64 => "m",
-            Base64pad => "M",
-            Base64url => "u",
-            Base64urlpad => "U",
+            Base1 => b'1',
```

Comment:
if we use u8 how about directly storing them in the enum and using casting to u8 instead of  code and TryFrom<u8> instead of from_code?